### PR TITLE
backport 6.x: doc: update file.data keyword documentation v1 6563

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

https://redmine.openinfosecfoundation.org/issues/6563

Describe changes:
- backport of file.data keyword doc updates
-
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Review by victorjulien — APPROVED** [review]

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -787,3 +787,4 @@ Notes
    pattern '<html' is absent from the first inspected chunk.
 
 -  ``file_data`` can also be used with SMTP
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:789]: A bit curious about why this wasn't removed here, but was for the other two Suri versions. Other than that, looks good to me!

Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:789]: A bit confused wrt why this was left here, but removed from 7 and 8 documentation.

**Review by jufajardini — COMMENTED** [review]
While reviewing our merging PR, I noticed a difference between the documentation for `file.data` in 6 and the ones for 7 and 8. Left an inline comment about it :)

**Review by inashivb — CHANGES_REQUESTED** [review]
Missing the `cherry-picked from` line in commit message needed to correlate backports. See https://github.com/OISF/suricata/commit/24404135e9b6cf7bf889ef831769402e55e7b720

It can be done with `git cherry-pick -x <commit-hash>` while cherry picking a commit.

**Comment by victorjulien** [comment]
> Missing the `cherry-picked from` line in commit message needed to correlate backports. See [2440413](https://github.com/OISF/suricata/commit/24404135e9b6cf7bf889ef831769402e55e7b720)
> 
> It can be done with `git cherry-pick -x <commit-hash>` while cherry picking a commit.

I don't think this is critical for doc backports, esp contributed ones. And especially since we're not documenting this anywhere. See also https://redmine.openinfosecfoundation.org/issues/6568

File: doc/userguide/rules/http-keywords.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:789]: It looks like I missed this part when I was fixing up the cherry-pick conflicts (I also apparently forgot the -x). I can get this fixed up.

Comment (jufajardini) [reply doc/userguide/rules/http-keywords.rst:789]: I see, thanks! I'll leave the decision to Victor in this, but IMHO a fixup works.

**Comment by jmtaylor90** [comment]
continued in https://github.com/OISF/suricata/pull/9867

**Pull request closed without merging**